cli/cmd/doc.go

```diff
@@ -0,0 +1,91 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+	"sigs.k8s.io/yaml"
+)
+
+type cmdOption struct {
+	Name         string
+	Shorthand    string `yaml:",omitempty"`
+	DefaultValue string `yaml:"default_value,omitempty"`
+	Usage        string `yaml:",omitempty"`
+}
+
+type cmdDoc struct {
+	Name             string
+	Synopsis         string      `yaml:",omitempty"`
+	Description      string      `yaml:",omitempty"`
+	Options          []cmdOption `yaml:",omitempty"`
+	InheritedOptions []cmdOption `yaml:"inherited_options,omitempty"`
+	Example          string      `yaml:",omitempty"`
+	SeeAlso          []string    `yaml:"see_also,omitempty"`
+}
+
+var cmdList []cmdDoc
+
+func newCmdDoc() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:    "doc",
+		Hidden: true,
+		Short:  "Generate YAML documentation for the Linkerd CLI",
+		Args:   cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := generateDocs(RootCmd); err != nil {
+				return err
+			}
+
+			out, err := yaml.Marshal(cmdList)
+			if err != nil {
+				return err
+			}
+
+			fmt.Println(fmt.Sprintf("%s", out))
```

Comment:
Super minor, but I'd typically write this as:

```suggestion
			fmt.Println(string(out))
```